dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBNoSQL/AzureCosmosDBNoSQLVectorStoreRecordCollection.cs

```diff
@@ -0,0 +1,529 @@
+            {
+                yield return key;
+            }
+        }
+    }
+
+    #region
+
+    private async Task<T> RunOperationAsync<T>(string operationName, Func<Task<T>> operation)
+    {
+        try
+        {
+            return await operation.Invoke().ConfigureAwait(false);
+        }
+        catch (Exception ex)
+        {
+            throw new VectorStoreOperationException("Call to vector store failed.", ex)
+            {
+                VectorStoreType = DatabaseName,
+                CollectionName = this.CollectionName,
+                OperationName = operationName
+            };
+        }
+    }
+
+    private async Task RunOperationAsync(string operationName, Func<Task> operation)
+    {
+        try
+        {
+            await operation.Invoke().ConfigureAwait(false);
+        }
+        catch (Exception ex)
+        {
+            throw new VectorStoreOperationException("Call to vector store failed.", ex)
+            {
+                VectorStoreType = DatabaseName,
+                CollectionName = this.CollectionName,
+                OperationName = operationName
+            };
+        }
+    }
+
+    private ContainerProperties GetContainerProperties()
+    {
+        var embeddings = new Collection<Embedding>();
+        var vectorIndexes = new Collection<VectorIndexPath>();
+
+        foreach (var property in this._vectorStoreRecordDefinition.Properties.OfType<VectorStoreRecordVectorProperty>())
+        {
+            var vectorPropertyName = this._storagePropertyNames[property.DataModelPropertyName];
+
+            if (!property.Dimensions.HasValue)
+            {
+                throw new VectorStoreOperationException($"Cannot create a collection, {nameof(property.Dimensions)} should be specified for {vectorPropertyName} vector property.");
+            }
+
+            var path = $"/{vectorPropertyName}";
+
+            var embedding = new Embedding
+            {
+                DataType = GetDataType(property.PropertyType, vectorPropertyName),
+                Dimensions = (ulong)property.Dimensions,
+                DistanceFunction = GetDistanceFunction(property.DistanceFunction, vectorPropertyName),
+                Path = path
+            };
+
+            var vectorIndexPath = new VectorIndexPath
+            {
+                Type = GetIndexKind(property.IndexKind, vectorPropertyName),
+                Path = path
+            };
+
+            embeddings.Add(embedding);
+            vectorIndexes.Add(vectorIndexPath);
+        }
+
+        var vectorEmbeddingPolicy = new VectorEmbeddingPolicy(embeddings);
+        var indexingPolicy = new IndexingPolicy { VectorIndexes = vectorIndexes };
+
+        return new ContainerProperties(this.CollectionName, partitionKeyPath: $"/{this._partitionKeyStoragePropertyName}")
+        {
+            VectorEmbeddingPolicy = vectorEmbeddingPolicy,
+            IndexingPolicy = indexingPolicy
+        };
```

Comment:
Since we are defining our own indexing policy, the default policy which includes all fields doesn't apply.  I think this is correct, since we allow people to selectively index Data fields using the IsFilterable and IsSearchable settings.
We will need to support that here as well though for Data fields.  We probably want to add a path for each field that is filterable using this type of format:
https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/how-to-manage-indexing-policy?tabs=dotnetv3%2Cpythonv3#opt-in-policy-to-selectively-include-some-property-paths

Reply:
If we will rely on `IsFilterable` or `IsSearchable` parameters by default, users will be forced to manually select which field to index every time. While we can preserve default logic by setting `Automatic` field to `true`, which is default behavior, together with vectors configuration:
```csharp
var indexingPolicy = new IndexingPolicy { VectorIndexes = vectorIndexes, Automatic = true };
```
![image](https://github.com/user-attachments/assets/f4747472-c265-4660-b3e5-f2b866f3812b)

What we can also do - we can add a boolean flag to `CollectionOptions` class to specify whether to use default `Automatic` indexing policy or apply indexing policy for each field separately by relying on `IsFilterable` or `IsSearchable` properties, in this case we can support both options. Let me know what you think about it.

Reply:
Added indexing policy handling for Data fields.